### PR TITLE
fix: allow the agent to accept multiple versions of legacy NR distributed tracing headers

### LINKED ISSUE #863: Agent should be flexible re: case sensitivity of incoming DT header
This issue is a follow-on action from #652.

Per the agent specs (NR internal-only link: https://source.datanerd.us/agents/agent-specs/blob/main/distributed_tracing/New-Relic-Payload.md), agents should accept inbound an inbound distributed tracing header with the key cased as any of the following: `newrelic`, `NEWRELIC`, `Newrelic`.

Currently, the full agent's internal method that looks for an inbound NR DT header only looks for `newrelic`: https://github.com/newrelic/newrelic-dotnet-agent/blob/686a119865cdfb382248828ee41bd7606bacb102/src/Agent/NewRelic/Agent/Core/DistributedTracing/TracingState.cs#L245

We should update this method to be compliant with the agent specs, along with appropriate unit and possibly integration tests.

There is also a constant defined in the API namespace that has the header key listed as `Newrelic`: https://github.com/newrelic/newrelic-dotnet-agent/blob/2ba99ea91ac20f79af40259e485949c6b3b1aac3/src/Agent/NewRelic.Api.Agent/Constants.cs#L31  This constant appears to be unreferenced anywhere in the agent solution.  It might have been put there for users of the old, obsoleted manual DT APIs `Insert/AcceptDistributedTracePayload()`.  The replacement `Insert/AcceptDistributedTraceHeaders()` APIs do not require this constant.  It should be removed for general code cleanliness and also to avoid confusing users.

### PR DESCRIPTION
**Pull request by nr-ahemsath**
## Description

Resolves #863.  The agent will now accept incoming legacy distributed tracing headers where the key is any of the following: `newrelic`, `Newrelic`, `NEWRELIC`.

There were some performance concerns associated with doubling the number of potential `getter` calls when there are no tracing headers present, such as the initial request to a web front-end application if the browser is not instrumented.  I did the following test:

Created standard ASP.Net Core web api app (`dotnet new webapi`), instrumented with agent (either latest release or this branch)
Made requests to http://localhost:5000/WeatherForecast as fast as possible for five minutes and counted the number of requests completed in that time.
Used `dotnet-counters` to collect other performance data (CPU/memory/GC, etc.)
Results:
Main branch: 
1277099 requests completed
PR branch:
1265241 requests completed
No significant difference in CPU, memory, etc.

This PR also removes dead code associated with the obsoleted `AcceptDistributedTracePayload` API. 🧹 

# Author Checklist
- [X] Unit tests, Integration tests, and Unbounded tests completed
- [X] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by chynesNR** [comment]
The changes look good, but I'm a little wary of removing API methods without a major rev. It's true they've been deprecated for years, and anyone using them has been getting Warnings, but it's possible someone left them in and has been ignoring them. I'll go with the majority, though, if they think it's low-risk.

File: src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core.Metric/ApiSupportabilityMetricCounters.cs [hunk 1]
```diff
@@ -11,35 +11,32 @@ namespace NewRelic.Agent.Core.Metric
...
-        GetLinkingMetadata = 20,
-        TransactionAddCustomAttribute = 21,
-        TransactionGetCurrentSpan = 22,
-        SpanAddCustomAttribute = 23,
-        InsertDistributedTraceHeaders = 24,
-        AcceptDistributedTraceHeaders = 25,
-        SpanSetName = 26,
-        SetErrorGroupCallback = 27,
-        SetUserId = 28
+        CurrentTransaction = 0,
```
[the remark this answers is not in the document]
Comment (chynesNR) [reply src/Agent/NewRelic/Agent/Core/NewRelic.Agent.Core.Metric/ApiSupportabilityMetricCounters.cs:14]: Good question! I was concerned about that, too, but this is only used for Supportability metrics, not the actual API calls, and the values are only used as array indices that point back to the method name.

**Comment by tippmar-nr** [comment]
Regarding removing the deprecated public APIs, I don't have enough history to understand how long they've been "unused." It would be unfortunate to blow a customer out of the water if they happened to still be using the APIs, in spite of all the warnings against it.

But... 

It feels like the changes that remove the APIs are actually bigger than (and, unless I'm missing something, unrelated to) the bug that is being resolved with this PR.  To that end, I think I'd prefer to split the work out into two separate PRs.

**Comment by nr-ahemsath** [comment]
> The changes look good, but I'm a little wary of removing API methods without a major rev. It's true they've been deprecated for years, and anyone using them has been getting Warnings, but it's possible someone left them in and has been ignoring them. I'll go with the majority, though, if they think it's low-risk.

If you look at https://github.com/newrelic/newrelic-dotnet-agent/blob/a5df0829b6e29ab2c656ddb459c26cfc68d0af51/src/Agent/NewRelic/Agent/Core/Api/TransactionBridgeApi.cs#L57 you'll see that since 9.0, anybody calling that API from their application just gets a warning and nothing functional happens.

**Comment by nr-ahemsath** [comment]
> Regarding removing the deprecated public APIs, I don't have enough history to understand how long they've been "unused." It would be unfortunate to blow a customer out of the water if they happened to still be using the APIs, in spite of all the warnings against it.

See my reply to @chynesNR's similar question.  The warning will remain in place so there should be no functional change to customers still (inadvertently, we'd hope) using the obsoleted API.

**Review by tippmar-nr — APPROVED** [review]

**Review by chynesNR — APPROVED** [review]

**Pull request merged**